packages/grid/_modules_/grid/utils/EventEmitter.ts

```diff
@@ -0,0 +1,68 @@
+type Listener = (...args: any[]) => void;
+
+// Used https://gist.github.com/mudge/5830382 as a starting point.
+// See https://github.com/browserify/events/blob/master/events.js for
+// the Node.js (https://nodejs.org/api/events.html) polyfill used by webpack.
+export class EventEmitter {
+  maxListeners = 10;
+
+  warnOnce = false;
+
+  events: { [key: string]: Listener[] } = {};
+
+  on(eventName: string, listener: Listener): void {
+    if (!Array.isArray(this.events[eventName])) {
+      this.events[eventName] = [];
+    }
+
+    this.events[eventName].push(listener);
+
+    if (this.events[eventName].length > this.maxListeners && this.warnOnce === false) {
+      this.warnOnce = true;
+      console.warn(
+        [
+          `Possible EventEmitter memory leak detected. ${this.events[eventName].length} ${eventName} listeners added.`,
```

Comment:
Well it's not because you add a lot of listeners that there is a leak...

Reply:
The logic match Node.js. The limit can be configured. We (maintainer) will increase the limit in the future as we add more listeners if the use cases are legit.

Reply:
Ok so we need to hold on this one as I see warning everywhere with more than 10 listeners on the state PR

Reply:
Oh, nice! How can I reproduce it? I have opened the storybook but I don't see any.

Reply:
The problem comes from ColumnHeaderItem that calls `useApiEventHandler`. I have removed the memo of this component in the state refact, and I get the warnings which makes sense.
The event should be attached to the parent and not the item
I have refactored it

Reply:
Oh nice. So you tried the logic in your local branch. Does it mean the the polyfill of webpack v4 isn't raising any warnings? :o